dts/bindings/pinctrl/nxp,kinetis-pinmux.yaml

```diff
@@ -5,28 +5,20 @@ version: 0.1
 description: >
     This is a representation of the Kinetis Pinmux node
 
+inherits:
```

Comment:
I'm also having trouble understanding how "inherits" works right now. This makes it difficult for me to understand these changes. In particular, I don't understand the schema/semantics for bindings and how it relates to their inheritance structure.

In addition to reading extract_dts_includes.py and playing with it in the debugger, I've read this documentation: http://docs.zephyrproject.org/devices/dts/device_tree.html

I see how defines can get generated from a node's name (along with it's parent's names and addresses), and how the "compatible" constraint and "generation: define" control that.

But there are other things that are less clear to me here especially where inheritance is concerned. For example, the ADC binding for nxp,kinetis-adc16.yaml inherits from adc.yaml in the same way this file is being changed:

https://github.com/zephyrproject-rtos/zephyr/blob/master/dts/bindings/iio/adc/nxp%2Ckinetis-adc16.yaml

However, looking at the base adc.yaml (which corresponds to the pinmux.yaml being introduced in this patch), it defines a "category: required" clocks property:

https://github.com/zephyrproject-rtos/zephyr/blob/master/dts/bindings/iio/adc/adc.yaml

This is confusing because when looking at the dts for frdm_k64f, which has ADC nodes, those same nodes don't have any clocks property (I checked the build/zephyr/frdm_k64f.dts_compiled output as well just to make sure it wasn't in a magic overlay or something):

https://github.com/zephyrproject-rtos/zephyr/blob/master/boards/arm/frdm_k64f/frdm_k64f.dts
https://github.com/zephyrproject-rtos/zephyr/blob/master/dts/arm/nxp/nxp_k6x.dtsi

I would have expected extract_dts_includes.py to raise an error there, since it looks like it's trying to define a required property which is missing, but that doesn't seem to be happening, so I'm not sure what the relationship between parent and child is there (especially since I'm having trouble following extract_dts_includes.py in general).

I did read device_node.yaml.template as well, but that seems to be more descriptive than really enforcing a schema (the way that the pykwalify library is used with the sanitycheck test cases to enforce that the actual yaml files match the schema).

Reply:
@mbolivar: You're totally right. I think your understanding is correct, problem is that field is not checked. I raised following issue to address this point: #5460
Don't hesitate to comment on this issue.